### DISCUSSION TITLE
Can unshareable stages' session layers be saved?

### DISCUSSION THREAD

**Discussion by pierre-ed** [Autodesk/maya-usd#2352] [Q&A]
Hello everyone!

Here I am, with yet another question regarding `mayaUsdProxyShape` nodes that do **not** share their respective stage.   

We did not notice it in our first tests since when we uncheck `shareStage`, the generated root layer automatically becomes the new stage's edit target, but it appears that the session layer of an unshared stage cannot be serialized in the Maya scene (or in any separate USD layer - we do not get the usual prompt asking us for the saving strategy to adopt).   
And naturally, all of its edits are simply lost once the scene is closed.   
![20220513__autodesk__session_layer_not_serialized](https://user-images.githubusercontent.com/5444972/168341307-ddb2b560-1cc1-46be-95b0-504fb9085384.gif)

This behavior definitely diverges from what happens with shared stages' session layers, which can be saved as string attributes for the `mayaUsdLayerManager` node present in the scene. Is there a specific reason behind this discrepancy? Or are we simply doing something wrong?
Once again, thanks for your help.
   --Pierre

#### Specs

- OS & version: CentOS Linux release 7.6.1810 (Core)
- Compiler & version: gcc (GCC) 6.3.1 20170216
- Maya version: Maya 2022.3
- Maya USD: 0.17.0
- Pixar USD: 21.11

**Comment by womanyoyoyo-adsk** [comment]
Hi @pierre-ed,

I've reproduced this on our side and logged it as MAYA-123361. Looks like we are not serializing the Unshareable stage's session layer into the Maya file. 

I have found a workaround. When I opened the Maya file and checked 'Shareable' the edits are in the session layer. See here:

![sessionLayerUnshareable](https://user-images.githubusercontent.com/49446314/168847512-6c83d7c0-1b42-4af5-b385-4c3ecae7a3b1.gif)

When in an Unshareable stage and trying to save the scene, the Save prompt should appear with option 1 disabled. I haven't logged this yet but I will update this thread when I do.

Hope this helps for now. 
Natalia
